service/src/main/java/bio/terra/workspace/common/utils/ControllerValidationUtils.java

```diff
@@ -115,4 +117,23 @@ public static String samCreateAction(ControlledResourceFields commonFields) {
             commonFields.getAccessScope(), commonFields.getManagedBy())
         .getSamCreateResourceAction();
   }
+
+  /**
+   * Validate the format of an ipAddress or range of addresses for Azure SAS tokens. We can't do
+   * this directly in the generated spring code yet, because the swagger codegen plugin doesn't
+   * support the use of oneOf in schema generation.
+   *
+   * @param ipRange a single ip address, or a range of ip addresses separated by a dash ('-')
+   */
+  public static void validateIpAddressRange(@Nullable String ipRange) {
```

Comment:
There's a couple of edge cases that aren't tested here, like out of order ranges (like "168.1.5.70-168.1.5.60") and mixing ipv4 and ipv6.

Does it make sense to add those cases, or does it run too much of a risk of catching valid input?

Reply:
I guess I'm curious about what Azure will do if those edge cases are allowed through. Will Azure just return an error? Will it ignore the range entirely and return an unrestricted SAS token? Some other unexpected behavior?